R/raincloudPlots.R

```diff
@@ -0,0 +1,908 @@
+        "Interval around mean represents ",
+        options$meanCiWidth * 100,
+        "% confidence interval;<br>confidence intervals were computed independently for each group.")
+    } else {
+      meanInterval <- NULL
+    }
+  } else {
+    meanInterval <- NULL
+  }
+
+  jitter <- if (options$jitter) {
+    gettextf("Points are slightly jittered from their true values.")
+  } else {
+    NULL
+  }
+
+  warningAxisLimits <- if (warningAxisLimits) {
+    gettextf("<span style = 'color: darkorange'> Warning! Some data is not shown<br>because it lies outside of the interval set by the custom axis limits.</span>")
+  } else {
+    NULL
+  }
+
+  errorVioSides <- if (errorVioSides && options$customSides) {
+    gettextf("<span style = 'color: darkorange'> Error with custom orientation: Specified number of clouds does not match clouds in plot.<br>Reverted to default all 'R'. Point nudge set to 0.</span>")
+  } else {
+    NULL
+  }
+
+  output <- paste0(exclusions, "\n\n", meanInterval, "\n\n", jitter, "\n\n", warningAxisLimits, "\n\n", errorVioSides)
+  return(output)
+}  # End .rainCaption()
+
+
+
+# .rainCreateTables ----
+.rainCreateTables <- function(jaspResults, dataInfo, options, ready) {
+
+  if (!options$table) return()
+
+  # Create container in jaspResults
+  if (is.null(jaspResults[["containerTables"]])) {
+
+    jaspResults[["containerTables"]] <- createJaspContainer(title = gettext("Statistics"))
+    jaspResults[["containerTables"]]$dependOn(
+      c(
+        "dependentVariables", "table", "tableBoxStatistics"
+      )
+    )
+  }  # End create container
+
+  # Access through container object
+  container <- jaspResults[["containerTables"]]
+
+  # Placeholder table, if no dependentVariables
+  if (!ready) {
+    placeholderTable <- createJaspTable(title = "", dependencies = c("dependentVariables"))
+    placeholderTable <- .rainAddTableColumns(options, placeholderTable)
+    container[["placeholderTable"]] <- placeholderTable
+    return()
+  }
+
+  # Table for each dependentVariable
+  for (variable in options$dependentVariables) {
+
+    # If table for variable already exists, we can skip recreating table
+    if (!is.null(container$variable)) next
+
+    variableTable <- createJaspTable(title = variable)
+    variableTable$dependOn(optionContainsValue = list(dependentVariables = variable))  # Depends on respective variable
+    variableTable <- .rainAddTableColumns(options, variableTable)
+
+    .rainFillTable(jaspResults, dataInfo, variable, options, variableTable)
+
+    container[[variable]] <- variableTable
+
+  }  # End of for loop
+
+}  # End .rainCreateTables()
+
+
+
+# .rainAddTableColumns() ----
+.rainAddTableColumns <- function(options, inputTable) {
+
+  tableInProgress <- inputTable
+
+  # Add columns; names of boxplot statistics are like extracted from ggplot, see boxData and .rainFillTable
+  tableInProgress$addColumnInfo(name = "primaryFactor", title = "Primary Factor", type = "string")
```

Comment:
```suggestion
  tableInProgress$addColumnInfo(name = "primaryFactor", title = gettext("Primary Factor"), type = "string")
```
same for all other titles of the table columns.